Content.Client/Doors/FirelockSystem.cs

```diff
@@ -0,0 +1,30 @@
+using Content.Shared.Doors.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.ResourceManagement;
+using Robust.Shared.Timing;
+
+namespace Content.Client.Doors;
+
+public sealed class FirelockSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] protected readonly SharedAppearanceSystem AppearanceSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<FirelockComponent, AppearanceChangeEvent>(OnAppearanceChange);
+    }
+
+    private void OnAppearanceChange(EntityUid uid, FirelockComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null || !_gameTiming.IsFirstTimePredicted)
```

Comment:
Appearance updates only happen during frame-updates, so there is no need to check `IsFirstTimePredicted`.